DCRack/dc_rack_server.py

```diff
@@ -300,42 +367,38 @@ def pushRegistryValue(self, dac, slow, voltage):
             num = 0
         elif num < -2.5:
             num = -2.5
+
+        # convert voltage into 16-bit number, plus low bit for DAC selection
         if dac:
-            intNum = long(float(num+2.5)/5.0 * 65535)
-            intNum = (intNum << 1)|1
-            if slow:
-                intNum = intNum|131702
+            dac_value = long(float(num+2.5)/5.0 * 0xffff)
+            reg = (dac_value << 1) | 1
         else:
-            intNum = long(float(num)/2.5 * 65535)
-            intNum = (intNum << 1)
-            if slow:
-                intNum = intNum|131702
-                
-        #Push bits to proper positions
-        Byte1 = long(intNum)
-        Byte2 = long(intNum>>6)
-        Byte3 = long(intNum>>12)
-
-        #Write 8 bit sequences
-        yield self.write([128|(Byte3&63)])
-        yield self.write([128|(Byte2&63)])
-        yield self.write([128|(Byte1&63)])
-        
+            dac_value = long(float(num)/2.5 * 0xffff)
+            reg = (dac_value << 1)
+
+        # set high bit for slew rate
+        if slow:
+            reg |= 0x20000
```

Comment:
There was a nasty bug here before. The expression `intNum | 131702` was meant to set the 18th bit for the slew rate, but `131702 == 0x20276`. It should have been `131072 == 0x20000`. So we were previously messing up the requested DAC levels by turning on a bunch of bits that weren't requested. The moral of the story is never use decimal constants when doing bit manipulations; always use hex or, if the language allows, binary.